src/sink/blocking.rs

```diff
@@ -0,0 +1,114 @@
+use {Async, AsyncSink};
+use sink::Sink;
+use executor;
+use task_impl::ThreadNotify;
+
+/// Provides thread-blocking operations on a sink.
+///
+/// See [`blocking`](fn.blocking.html) documentation for more details.
+#[derive(Debug)]
+#[must_use = "sinks do nothing unless used"]
+pub struct Blocking<T: Sink> {
+    inner: Option<executor::Spawn<T>>,
```

Comment:
I can't figure out why this is an `Option`; AFAICT it's always `Some` (and everything except the dtor asserts that).

Reply:
It is an `Option` to support `into_inner` (which has to be able to extract the inner type without triggering the dtor).

Reply:
Ah yep, missed that. :+1: